File: detox/src/android/espressoapi/web/WebElement.js
/**

	This code is generated.
	For more information see generation/README.md.
*/



class WebElement {
  static tap(element) {
    return {
      target: element,
      method: "tap",
      args: []
    };
  }

  static typeText(element, text) {
    if (typeof text !== "string") throw new Error("text should be a string, but got " + (text + (" (" + (typeof text + ")"))));
    return {
      target: element,
      method: "typeText",
      args: [text]
    };
  }

  static replaceText(element, text) {
    if (typeof text !== "string") throw new Error("text should be a string, but got " + (text + (" (" + (typeof text + ")"))));
    return {
      target: element,
      method: "replaceText",
      args: [text]
    };
  }

  static clearText(element) {
    return {
      target: element,
      method: "clearText",
      args: []
    };
  }

  static scrollToView(element) {
    return {
      target: element,
      method: "scrollToView",
      args: []
    };
  }

  static getText(element) {
    return {
      target: element,
      method: "getText",
      args: []
    };
  }

  static runScript(element, script) {
    if (typeof script !== "string") throw new Error("script should be a string, but got " + (script + (" (" + (typeof script + ")"))));
    return {
      target: element,
      method: "runScript",
      args: [script]
    };
  }

  static runScriptWithArgs(element, script, args) {
    if (typeof script !== "string") throw new Error("script should be a string, but got " + (script + (" (" + (typeof script + ")"))));

    if (typeof args !== 'object' || !args instanceof Array) {
      throw new Error('args must be an array, got ' + typeof args);
    }

    return {
      target: element,
      method: "runScriptWithArgs",
      args: [script, {
        type: "ArrayList<Object>",
        value: args
      }]
    };
  }

  static getCurrentUrl(element) {
    return {
      target: element,
      method: "getCurrentUrl",
      args: []
    };
  }

  static getTitle(element) {
    return {
      target: element,
      method: "getTitle",
      args: []
    };
  }

}

module.exports = WebElement;

File: detox/android/detox/src/full/java/com/wix/detox/espresso/web/WebElement.java
package com.wix.detox.espresso.web;

import androidx.test.espresso.web.model.Atom;
import androidx.test.espresso.web.model.Atoms;
import androidx.test.espresso.web.model.ElementReference;
import androidx.test.espresso.web.model.Evaluation;
import androidx.test.espresso.web.model.SimpleAtom;
import androidx.test.espresso.web.sugar.Web;
import androidx.test.espresso.web.webdriver.DriverAtoms;

import java.util.ArrayList;
import java.util.List;

public class WebElement {

    final WebViewElement webViewElement;
    final Atom<List<ElementReference>> matcherAtom;
    final int index;

    WebElement(WebViewElement webViewElement, Atom<List<ElementReference>> matcherAtom, int index) {
        this.webViewElement = webViewElement;
        this.matcherAtom = matcherAtom;
        this.index = index;
    }

    Web.WebInteraction<Void> getWebViewInteraction() {
        return webViewElement.webViewInteraction;
    }

    ElementReference get() {
        List<ElementReference> elements = getWebViewInteraction().perform(matcherAtom).get();

        if (elements == null || elements.size() == 0 || index >= elements.size()) {
            throw new RuntimeException(String.format("element was not found at index: %d", index));
        }

        return elements.get(index);
    }

    public void tap() {
        getWebViewInteraction().withElement(get()).perform(DriverAtoms.webClick());
    }

    public void typeText(String text) {
        getWebViewInteraction().withElement(get()).perform(DriverAtoms.webKeys(text));
    }

    public void replaceText(String text) {
        clearText();
        getWebViewInteraction().withElement(get()).perform(DriverAtoms.webKeys(text));
    }

    public void clearText() {
        getWebViewInteraction().withElement(get()).perform(DriverAtoms.clearElement());
    }

    public boolean scrollToView() {
        return getWebViewInteraction().withElement(get()).perform(DriverAtoms.webScrollIntoView()).get();
    }

    public String getText() {
        return getWebViewInteraction().withElement(get()).perform(DriverAtoms.getText()).get();
    }

    public Object runScript(String script) {
        return getWebViewInteraction().withElement(get()).perform(new SimpleAtom(script)).get().getValue();
    }

    public Object runScriptWithArgs(String script, final ArrayList<Object> args) {
        return getWebViewInteraction().withElement(get()).perform(Atoms.scriptWithArgs(script, args)).get().getValue();
    }

    public String getCurrentUrl() {
        return getWebViewInteraction().withElement(get()).perform(Atoms.getCurrentUrl()).get();
    }

    public String getTitle() {
        return getWebViewInteraction().withElement(get()).perform(Atoms.getTitle()).get();
    }
}
